### PR TITLE
add page + functionality to handle non-US users in the Core Add-On

### LINKED ISSUE #297: Implement a page to show if outside of the U.S.
Rally is currently not supported outside of the U.S.. We can detect this from the core-addon (see #263).

This bug is for designing and implementing the page to point users outside of the U.S. to. Ryan provided the mock:

image

@hamilton this is a blocker for GTM. Is this something you can take?

### PR DESCRIPTION
**Pull request by hamilton**
Closes #297 

This PR does the following:
- implements the page telling non-US users that they can't sign up for Rally. This uses `store.updatePlatformEnrollment(false)`
- ~adds an API to the Core add-on called `uninstall-rally` that we can use to uninstall the Core Add-On, used on the non-US user splash page. I'm open to remove this API call; it feels a little bit risky to have an API like this.~
- ~adds a unit test for the `uninstall-rally` API call.~

Checklist for reviewer:

- [ ] The description should reference a bug or github issue, if relevant.
- [ ] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [ ] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [ ] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

File: src/App.svelte [hunk 1]
```diff
@@ -40,12 +41,12 @@
...
-            // reset to first run until Rally uninstalls itself.
-            firstRun = true;
           }}
         />
     {/if}
   {/if}
 {:else}
-  <div>Sorry, Rally is not supported in this locale.</div>
+  <NonUSSplashPage on:remove-extension={() => {
+    store.uninstallRally();
```
Comment (Dexterp37) [inline src/App.svelte:50]:
```suggestion
    store.updatePlatformEnrollment(false);
```

Can you kindly check if this works?

File: src/App.svelte [hunk 2]
```diff
@@ -40,12 +41,12 @@
       <Main
           on:leave-rally={() => {
             store.updatePlatformEnrollment(false);
-            // reset to first run until Rally uninstalls itself.
-            firstRun = true;
```
Comment (Dexterp37) [inline src/App.svelte:44]: Why was this removed?

File: core-addon/Core.js [hunk 3]
```diff
@@ -194,6 +194,11 @@ module.exports = class Core {
         return this._unenrollStudy(message.data.studyID);
       case "unenrollment":
         return this._unenroll();
+      case "uninstall-rally":
+        // If a user is uninstalling Rally because they do not qualify to be 
+        // on the platform (e.g. they are non-US), uninstall it for them on request.
+        browser.management.uninstallSelf({ showConfirmDialog: false });
+        return Promise.resolve();
```
Comment (Dexterp37) [inline core-addon/Core.js:201]:
```suggestion
        return browser.management.uninstallSelf({ showConfirmDialog: false });
```
That API returns a promise, we should return it back to the caller.

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
Stealing review!

File: core-addon/Core.js (hunk 3 above)
Comment (hamilton) [reply core-addon/Core.js:201]: So I had this originally, but it caused the unit test I added to timeout. Any ideas there?

Comment (Dexterp37) [reply core-addon/Core.js:201]: You probably need [this line](https://github.com/mozilla-rally/rally-core-addon/blob/3a145ca14bc141eb5bb82898e7508c04ecf61aeb/tests/core-addon/unit/Core.test.js#L222). I'd recommend going with what I suggested you in the other comment, it will save you from the need of writing the unit test at all :D

Comment (hamilton) [reply core-addon/Core.js:201]: Yeah, I think the test already has that line but still times out. It's a problem with the `browser.management.uninstallSelf` mock itself. Maybe they've made it so it really simulates what it does & never actually resolves! At any rate, if `updatePlatformEnrollment(false)` can be overloaded to just follow the uninstall path, that works for me too!

File: src/web-extension-api.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ async function sendToCore(port, type, payload) {
     "study-enrollment",
     "study-unenrollment",
     "unenrollment",
-    "update-demographics",
+    "update-demographics"
```
Comment (Dexterp37) [inline src/web-extension-api.js:29]: Can you kindly retain the trailing comma, here?

File: src/routes/non-eligible-splashes/NonUSUser.svelte [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                </div>
+                <div class="waitlist">
+                <a target="_blank" href="__BASE_SITE__/waitlist/">Join the Waitlist
+                    <ExternalLink /></a>
+                </div>
+            </div>
+        </section>
+    </Main>
+</Layout>
+{/if}
```
Comment (Dexterp37) [inline src/routes/non-eligible-splashes/NonUSUser.svelte:91]: nit: can you add a newline?

**Review by Dexterp37 — APPROVED** [review]
r+wc
